intranet/apps/eighth/models.py

```diff
@@ -928,6 +933,10 @@ def add_user(self, user, request=None, force=False, no_after_deadline=False):
                 acts = EighthActivity.restricted_activities_available_to_user(user)
                 if self.activity.id not in acts:
                     exception.Restricted = True
+            else:
```

Comment:
the else makes it only check if the activity isn't restricted because if the activity is restricted, a blacklist shouldn't be necessary

this shouldn't matter anyways, since if the activity is restricted the field will be disabled so exception.Blacklist will never be set to true